src/watch_test.ts

```diff
@@ -34,81 +33,26 @@ const fakeConfig: {
     ],
 };
 
-// Object replacing real Request object in the test
-class FakeRequest extends EventEmitter implements RequestResult {
-    pipe(stream: Duplex): void {}
-    abort() {}
-}
+const systemUnderTest = (): [nock.Scope] => {
+    const scope = nock(fakeConfig.clusters[0].server);
+    return [scope];
+};
 
 describe('Watch', () => {
-    describe('DefaultRequest', () => {
-        it('should resume stream upon http status 200', (done) => {
-            const defaultRequest = new DefaultRequest((opts) => {
-                const req = request(opts);
-                // to prevent request from firing we must replace start() method
-                // before "nextTick" happens
-                (<any>req).start = () => {};
-                (<any>req).resume = () => done();
-                return req;
-            });
-            const req = defaultRequest.webRequest({
-                uri: `http://${server}/testURI`,
-            });
-            req.on('error', done);
-            (<any>req).emit('response', {
-                statusCode: 200,
-            });
-        });
-
-        it('should handle non-200 error codes', (done) => {
```

Comment:
i think we could convert this test to still validate this chunk of error wrapping by emitting a non-200 code and catching it on the stream?:
```
await fetch(watchURL, requestInit)
            .then((response) => {
                if (response.status === 200) {
                    response.body.on('error', doneCallOnce);
                    response.body.pipe(stream);
                } else {
                    const error = new Error(response.statusText) as Error & {
                        statusCode: number | undefined;
                    };
                    error.statusCode = response.status;
                    throw error;
                }
            })
            .catch(doneCallOnce);
```
i see you have it in other tests, but this one was isolated to just that functionality

Reply:
test `should handle error from request stream` does exactly that, unless I'm missing something here